mindmeld/active_learning/data_loading.py

```diff
@@ -300,49 +304,59 @@ def sample_and_update(
         and newly_sampled_queries.
         Args:
             sampling_size (int): Number of elements to sample in the next iteration.
             confidences_2d (List[List[float]]): Confidence probabilities per element.
+                (3d for tagger tuning)
             confidences_3d (List[List[List[float]]]): Confidence probabilities per element.
             heuristic (Heuristic): Selection strategy.
             confidence_segments (Dict[(str, Tuple(int,int))]): A dictionary mapping
                 segments to run KL Divergence.
+            tuning_type (TuningType): Component to be tuned ("classifier" or "tagger")
         Returns:
             newly_sampled_queries_ids (List[int]): List of ids corresponding the newly sampled
                 queries in the QueryCache.
         """
 
-        params_rank_3d = {"confidences_3d": confidences_3d}
-        if confidence_segments:
-            params_rank_3d["confidence_segments"] = confidence_segments
-
-        ranked_indices_2d = (
-            heuristic.rank_3d(**params_rank_3d)
-            if confidences_3d
-            else heuristic.rank_2d(confidences_2d)
-        )
+        if tuning_type == TuningType.CLASSIFIER:
+            params_rank_3d = {"confidences_3d": confidences_3d}
+            if confidence_segments:
+                params_rank_3d["confidence_segments"] = confidence_segments
 
-        newly_sampled_indices = ranked_indices_2d[:sampling_size]
-        remaining_indices = ranked_indices_2d[sampling_size:]
+            ranked_indices_2d = (
+                heuristic.rank_3d(**params_rank_3d)
+                if confidences_3d
+                else heuristic.rank_2d(confidences_2d)
+            )
 
-        if entity_tuning:
-            ranked_entity_indices = heuristic.rank_entities(entity_confidences)
-            newly_sampled_indices_entity = ranked_entity_indices[:sampling_size]
-            remaining_indices_entity = ranked_entity_indices[sampling_size:]
+            newly_sampled_indices = ranked_indices_2d[:sampling_size]
+            remaining_indices = ranked_indices_2d[sampling_size:]
 
-            newly_sampled_indices = list(
-                set(newly_sampled_indices).union(newly_sampled_indices_entity)
-            )
-            remaining_indices = list(
-                set(remaining_indices).intersection(remaining_indices_entity)
-            )
+        else:
+            try:
+                ranked_entity_indices = heuristic.rank_entities(confidences_2d)
+            except (TypeError, ValueError):
+                # if heuristic does not have entity AL support default to entropy
+                heuristic = EntropySampling
+                ranked_entity_indices = heuristic.rank_entities(confidences_2d)
+
+            newly_sampled_indices = ranked_entity_indices[:sampling_size]
+            remaining_indices = ranked_entity_indices[sampling_size:]
+
+            # PLACEHOLDER CODE for joint tuning (deprecated for now).
```

Comment:
I think adding an additional script that reads in the json files created for intent/domain and entity for a run and generate a plot showing trends in overall app accuracy resolves this concern. Mostly because as discussed with @ritvikshrivastava, if multiple strategies are selected for each level, we've have to run this eval for all combinations.   

If @KKsharma99, @serapio are in agreement with this, let's remove this commented block of code.